File: Budget.Server/Entities/Transaction.cs
using Budget.Server.Enums;
using System.ComponentModel.DataAnnotations;

namespace Budget.Server.Entities
{
    public class Transaction
    {
        [Key]
        public int Id { get; set; }
        public TransactionType Type { get; set; } = TransactionType.None;
        //public Account Account { get; set; } -> Currency will be handled inside the Account entity
        public double Amount { get; set; } = 0;
        public string Title { get; set; } = string.Empty;
        public DateOnly? Date { get; set; }
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.None;
        public string Comment { get; set; } = string.Empty;

        // Merchant -> string or entity ?

        // public int? CategoryId { get; set; };
        // public Category? Category { get; set; };
    }
}

File: budget.client/src/models/Transaction.ts
import { TransactionType } from '@/enums/TransactionType.ts';
import { PaymentMethod } from '@/enums/PaymentMethod.ts';

interface Transaction {
    id: number;
    type: TransactionType;
    amount: number;
    title: string;
    date?: string;
    paymentMethod: PaymentMethod;
    comment: string;
}

export type { Transaction };

export const DefaultTransaction: Transaction = {
    id: 0,
    type: TransactionType.None,
    amount: 0,
    title: '',
    date: undefined,
    paymentMethod: PaymentMethod.None,
    comment: ''
};
